src/files/coverageparser.ts

```diff
@@ -17,147 +17,293 @@ export class CoverageParser {
      * @param files array of coverage files in string format
      */
-    public async filesToSections(files: Map<string, string>): Promise<Map<string, Section>> {
-        let coverages = new Map<string, Section>();
+    public async filesToSections(
+        files: Map<string, string>
+    ): Promise<Map<string, Section>> {
+        const coverages = new Map<string, Section>();
 
         for (const file of files) {
             const fileName = file[0];
             const fileContent = file[1];
 
-            // file is an array
-            let coverage = new Map<string, Section>();
-
             // get coverage file type
             const coverageFile = new CoverageFile(fileContent);
             switch (coverageFile.type) {
                 case CoverageType.CLOVER:
-                    coverage = await this.xmlExtractClover(fileName, fileContent);
+                    await this.xmlExtractClover(
+                        coverages,
+                        fileName,
+                        fileContent
+                    );
                     break;
                 case CoverageType.JACOCO:
-                    coverage = await this.xmlExtractJacoco(fileName, fileContent);
+                    await this.xmlExtractJacoco(
+                        coverages,
+                        fileName,
+                        fileContent
+                    );
                     break;
                 case CoverageType.COBERTURA:
-                    coverage = await this.xmlExtractCobertura(fileName, fileContent);
+                    await this.xmlExtractCobertura(
+                        coverages,
+                        fileName,
+                        fileContent
+                    );
                     break;
                 case CoverageType.LCOV:
-                    coverage = await this.lcovExtract(fileName, fileContent);
+                    this.lcovExtract(coverages, fileName, fileContent);
                     break;
                 default:
                     break;
             }
-
-            // add new coverage map to existing coverages generated so far
-            coverages = new Map([...coverages, ...coverage]);
         }
-
         return coverages;
     }
 
-    private async convertSectionsToMap(
-        data: Section[],
-    ): Promise<Map<string, Section>> {
-        const sections = new Map<string, Section>();
+    private async addSections(
+        coverages: Map<string, Section>,
+        data: Section[]
+    ): Promise<void[]> {
         const addToSectionsMap = async (section: Section) => {
-            sections.set(section.title + "::" + section.file, section);
+            const key = [section.title, section.file].join("::");
+            const existingSection = coverages.get(key);
+
+            if (!existingSection) {
+                coverages.set(key, section);
+                return;
+            }
+
+            const mergedSection = this.mergeSections(existingSection, section);
```

Comment:
[F] If we already have a section with the current title and file, then we merge the line and branch information as they are used downstream.